realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -85,13 +86,14 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
 
     // Realm path will be used as the key to store different RealmCaches. Different Realm configurations with same path
     // are not allowed and an exception will be thrown when trying to add it to the cache map.
+    // The RealmCache won't be removed from the map even when all Realm instances associated to it get closed. Instead,
```

Comment:
What case is optimised by this? Not that I expect people to open 1000's of Realms, but some apps might be opening quite a lot of different Realms (think synced chat app with a Realm pr. user/channel)